src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java

```diff
@@ -288,6 +288,40 @@ public DescriptorImpl() {
         
     }
 
+    @Issue("JENKINS-28822")
+    @Test public void interruptingAbortsBuild() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node {\n" +
+                "  timeout(time: 1, unit: 'SECONDS') {" +
+                (Functions.isWindows()
+                        ? "bat 'ping -n 6 127.0.0.1 >nul'\n"
+                        : "sh 'sleep 5'\n") +
+                "  }" +
+                "}", true));
+        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        j.waitForCompletion(b);
+        // Fails with Result.FAILURE as of commit a224295d12.
+        j.assertBuildStatus(Result.ABORTED, b);
```

Comment:
I could also add some assertions that show what exceptions are thrown if reviewers think that would be useful.